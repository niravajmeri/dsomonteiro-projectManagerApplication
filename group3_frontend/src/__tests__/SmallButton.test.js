import React from 'react';
import { shallow } from 'enzyme';
import SmallButton from "../components/button/SmallButton"


describe('SmallButton', () => {
    it('should be defined', () => {
        expect(SmallButton).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(
            <SmallButton name='button test' />

        );
        expect(tree).toMatchSnapshot();
    });
    it('should have a button value', () => {
        const tree = shallow(
          <SmallButton name='button test' />
        );
        expect(typeof(tree.find('.smallButton').getElement().props['className'])).toBe('string');
      });
  


})