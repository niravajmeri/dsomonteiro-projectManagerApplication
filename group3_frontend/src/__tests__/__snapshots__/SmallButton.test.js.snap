// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallButton should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SmallButton
    name="button test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Glyphicon
          bsClass="glyphicon"
          className="pencilIcon"
          glyph="glyphicon glyphicon-pencil"
/>,
        undefined,
      ],
      "className": "smallButton",
      "href": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "glyphicon",
          "className": "pencilIcon",
          "glyph": "glyphicon glyphicon-pencil",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Glyphicon
            bsClass="glyphicon"
            className="pencilIcon"
            glyph="glyphicon glyphicon-pencil"
/>,
          undefined,
        ],
        "className": "smallButton",
        "href": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "glyphicon",
            "className": "pencilIcon",
            "glyph": "glyphicon glyphicon-pencil",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
